### PR TITLE
FList caching and backoff strategy 

### LINKED ISSUE #108: `protected` service backoff in case of failure
In case of protected service failure, we need to `back off` on restart and don't get stuck in a restart loop of the service. give the system sometime to breath

### LINKED ISSUE #109: flist caching
when we restart a container with the same flist, we can use a cached flist instead of re-downloading

### PR DESCRIPTION
**Pull request by muhamadazmy**
Fixes #109 
Fixes #108

### CODE REVIEW

File: apps/plugins/zfs/plugin/flist.go [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+func getFList(src string) (FList, error) {
+	u, err := url.Parse(src)
+	if err != nil {
+		return nil, err
+	}
+
+	if u.Scheme == "file" || u.Scheme == "" {
+		return LocalFList(u.Path), nil
+	} else if u.Scheme == "http" || u.Scheme == "https" {
+		if u.Hostname() != HubHost {
```
Comment (zaibon) [inline apps/plugins/zfs/plugin/flist.go]: Why do we enforce this ?
Notice prevent the user to host their flist somewhere else.

File: base/mgr/job.go [hunk 2]
```diff
@@ -65,6 +67,11 @@ func newJob(command *pm.Command, factory ProcessFactory, hooks ...pm.RunnerHook)
 
 		registerPID: registerPID,
 		waitPID:     waitPID,
+		backoff: BackOff{
+			Delay:    2 * time.Second,
+			Multiply: 1.5,
+			Max:      5 * time.Minute,
```
Comment (zaibon) [inline base/mgr/job.go]: So this is going to be set by default for all processes ?

Comment (muhamadazmy) [reply base/mgr/job.go]: Only protected processes (this include redis, and libvirtd, etc ...)

File: apps/plugins/zfs/plugin/flist.go (hunk 1 above)
Comment (muhamadazmy) [reply apps/plugins/zfs/plugin/flist.go]: I want to grantee that a `'%s.md5' % flist_url` always exists. so only our hub has this. If u try now any flist from a different location (any http server) it probably won't have this unless it's running an instance of the hub.

Anyway, this limitation is more of a sanity check, that can be loosen later on

**Pull request merged**